evm/client/rpc_client.go

```diff
@@ -467,22 +411,22 @@ func (r *RPCClient) SubscribeToHeads(ctx context.Context) (ch <-chan *evmtypes.H
 			if isHealthCheckRequest {
 				ctx = multinode.CtxAddHealthCheckFlag(ctx)
 			}
-			return r.latestBlock(ctx)
+			return r.LatestBlock(ctx)
```

Comment:
Can't we use adaptor's SubscribeToHeads instead of creating a new poller?

Reply:
Yes, and actually we don't need to override `SubscribeToFinalizedHeads` at all.